src/components/ImageView/ChannelMapView/ChannelMapViewComponent.tsx

```diff
@@ -0,0 +1,410 @@
+                    height={Math.floor(imageRenderHeight) + overlayStore.paddingBottom}
+                    type={"channel-map-inner"}
+                />
+            );
+        };
+        let width = Math.floor(channelMapViewWidth / channelMapStore.numColumns);
+        let height = Math.floor(channelMapViewHeight / channelMapStore.numRows);
+
+        if (column === 0) {
+            thisIs = "left";
+            width += overlayStore.paddingLeft;
+            overlayComponentLeft -= overlayStore.paddingLeft;
+        } else if (row === channelMapStore.numRows - 1 || row === lastRow || (row === lastRow - 1 && column > columnOfLastFrame)) {
+            thisIs = "bottom";
+            height += overlayStore.paddingBottom;
+        } else {
+            thisIs = "inner";
+        }
+        const getRef = (ref: HTMLCanvasElement) => {
+            if (ref) {
+                ref.id = `${column}_${row}`;
+                canvasRef.current = ref;
+            }
+        };
+
+        const draw = () => {
+            const canvas = canvasRef.current;
+            if (canvas && overlayComponentRef?.canvas) {
+                const ctx = canvas.getContext("2d");
+                if (ctx) {
+                    requestAnimationFrame(() => {
+                        const pixelRatio = devicePixelRatio * AppStore.Instance.imageRatio;
+                        canvas.width = width * pixelRatio;
+                        canvas.height = height * pixelRatio;
+                        const destCanvas = canvas.getContext("2d", {willReadFrequently: true});
+                        const w = overlayComponentRef.canvas.width;
+                        const h = overlayComponentRef.canvas.height;
+                        const destWidth = canvas.width;
+                        const destHeight = canvas.height;
+                        const cornerPaddingLeft = overlayStore.paddingLeft * pixelRatio;
+                        const cornerPaddingBottom = overlayStore.paddingBottom * pixelRatio;
+                        destCanvas.clearRect(0, 0, canvas.width, canvas.height);
+                        if (thisIs === "left") {
+                            destCanvas.drawImage(overlayComponentRef.canvas, 0, 0, w, h - cornerPaddingBottom, 0, 0, destWidth, destHeight);
+                        } else if (thisIs === "bottom") {
+                            destCanvas.drawImage(overlayComponentRef.canvas, cornerPaddingLeft, 0, w - cornerPaddingLeft, h, 0, 0, destWidth, destHeight);
+                        } else if (thisIs === "inner") {
+                            destCanvas.drawImage(overlayComponentRef.canvas, cornerPaddingLeft, 0, w - cornerPaddingLeft, h - cornerPaddingBottom, 0, 0, destWidth, destHeight);
+                        }
+                    });
+                }
+            }
+        };
+
+        const throttledDraw = _.throttle(draw, 50);
+
+        React.useEffect(() => {
+            throttledDraw();
```

Comment:
I think we can avoid using `overlayComponentRef`+throttling+`useEffect`(with a long list of dependencies) by setting the draw function as a props of `OverlayStore` and call it directly after the drawing of the corner overlay, but perhaps we can consider refactoring this in a separate PR in the next cycle since this requires some changes.